addons/udes_stock/models/stock_picking.py

```diff
@@ -299,6 +299,14 @@ def action_done(self):
         # Prevent recomputing the batch stat
         batches = mls.mapped("picking_id.batch_id")
         next_pickings = self.mapped("u_next_picking_ids")
+
+        # If operation type does not allow multiple lots in a single transfer then check for number 
+        # of different lots in move lines and raise an error if more than one
+        if self.picking_type_id.u_restrict_multi_lot_pickings:
+            lots = mls.mapped("lot_id")
```

Comment:
When does lot_name get turned into a lot because if its after this we may have a bug.

Reply:
Will need to go look into that.

Reply:
So `lot_name` will get turned into `lot` if the lot doesn't already exist in the super call, so I will move this check after that.

Reply:
Have now changed this so that the check is carried out after the super call of `action_done` is run.